x-pack/plugins/security_solution/server/endpoint/services/agent/clients/crowdstrike/crowdstrike_agent_status_client.ts

```diff
@@ -0,0 +1,171 @@
+    const sortField = 'crowdstrike.host.last_seen';
+
+    const query = {
+      bool: {
+        must: [
+          {
+            bool: {
+              filter: [
+                {
+                  terms: {
+                    'crowdstrike.host.id': agentIds,
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      },
+    };
+
+    try {
+      const [searchResponse, allPendingActions] = await Promise.all([
+        esClient.search(
+          {
+            index: CROWDSTRIKE_AGENT_INDEX_PATTERN,
+            from: 0,
+            size: 10000,
+            query,
+            collapse: {
+              // TODO: check if we should use crowdstrike.cid instead
+              field: 'crowdstrike.host.id',
+              inner_hits: {
+                name: 'most_recent',
+                size: 1,
+                sort: [
+                  {
+                    [sortField]: {
+                      order: 'desc',
+                    },
+                  },
+                ],
+              },
+            },
+            sort: [
+              {
+                [sortField]: {
+                  order: 'desc',
+                },
+              },
+            ],
+            _source: false,
+          },
+          { ignore: [404] }
+        ),
+
+        getPendingActionsSummary(esClient, metadataService, this.log, agentIds),
+      ]).catch(catchAndWrapError);
+
+      const mostRecentAgentInfosByAgentId = searchResponse?.hits?.hits?.reduce<
+        Record<string, RawCrowdstrikeInfo>
+      >((acc, hit) => {
+        // TODO TC: check if we should use crowdstrike.cid instead
+        if (hit.fields?.['crowdstrike.host.id'][0]) {
+          acc[hit.fields?.['crowdstrike.host.id'][0]] =
+            hit.inner_hits?.most_recent.hits.hits[0]._source;
+        }
+
+        return acc;
+      }, {});
+
+      const agentStatuses = await this.getAgentStatusFromConnectorAction(agentIds);
+
+      return agentIds.reduce<AgentStatusRecords>((acc, agentId) => {
+        const agentInfo = mostRecentAgentInfosByAgentId[agentId].crowdstrike;
+
+        const agentStatus = agentStatuses[agentId];
+        const pendingActions = allPendingActions.find(
+          (agentPendingActions) => agentPendingActions.agent_id === agentId
+        );
+
+        acc[agentId] = {
+          agentId,
+          agentType: this.agentType,
+          // TODO: check if we should use crowdstrike.cid instead
+          found: agentInfo?.host.id === agentId,
+          isolated: agentInfo?.status === CROWDSTRIKE_NETWORK_STATUS.CONTAINED,
+          lastSeen: agentInfo?.host.last_seen || '',
+          status:
+            agentStatus?.state === CROWDSTRIKE_STATUS_RESPONSE.ONLINE
+              ? HostStatus.HEALTHY
+              : // TODO TC: not sure what the UNKNOWN is - still to be figured
+              agentStatus?.state === CROWDSTRIKE_STATUS_RESPONSE.UNKNOWN
+              ? HostStatus.UNENROLLED
```

Comment:
If we don't know anything about the agent state at this point, then just assuming it is unenrolled doesn't seem right.  I would suggest you leave the fallback option as `offline` if we don't have an `Unknown` state.